plugins/Model_GAN/Model.py

```diff
@@ -0,0 +1,171 @@
+    inp = Input(shape=(input_size, input_size, nc_in))
+    #x = GaussianNoise(0.05)(inp)
+    x = conv_block_d(inp, 64, False)
+    x = conv_block_d(x, 128, False)
+    x = conv_block_d(x, 256, False)
+    out = Conv2D(1, kernel_size=4, kernel_initializer=conv_init, use_bias=False, padding="same", activation="sigmoid")(x)   
+    return KerasModel(inputs=[inp], outputs=out)
+
+def Encoder(nc_in=3, input_size=64):
+    inp = Input(shape=(input_size, input_size, nc_in))
+    x = Conv2D(64, kernel_size=5, kernel_initializer=conv_init, use_bias=False, padding="same")(inp)
+    x = conv_block(x,128)
+    x = conv_block(x,256)
+    x = conv_block(x,512) 
+    x = conv_block(x,1024)
+    x = Dense(1024)(Flatten()(x))
+    x = Dense(4*4*1024)(x)
+    x = Reshape((4, 4, 1024))(x)
+    out = upscale_ps(512)(x)
+    return KerasModel(inputs=inp, outputs=out)
+
+# Legacy, left for someone to try if interested
+#def Decoder(nc_in=512, input_size=8):
+#    inp = Input(shape=(input_size, input_size, nc_in))   
+#    x = upscale_block(inp, 256)
+#    x = Cropping2D(((0,1),(0,1)))(x)
+#    x = upscale_block(x, 128)
+#    x = res_block(x, 128)
+#    x = Cropping2D(((0,1),(0,1)))(x)
+#    x = upscale_block(x, 64)
+#    x = res_block(x, 64)
+#    x = res_block(x, 64)
+#    x = Cropping2D(((0,1),(0,1)))(x)
+#    x = Conv2D(3, kernel_size=5, kernel_initializer=conv_init, use_bias=False, padding="same")(x)
+#    out = Activation("tanh")(x)
+#    return KerasModel(inputs=inp, outputs=out)
+
+def Decoder_ps(nc_in=512, input_size=8):
+    input_ = Input(shape=(input_size, input_size, nc_in))
+    x = input_
+    x = upscale_ps(256)(x)
+    x = upscale_ps(128)(x)
+    x = upscale_ps(64)(x)
+    x = res_block(x, 64)
+    x = res_block(x, 64)
+    #x = Conv2D(4, kernel_size=5, padding='same')(x)   
+    alpha = Conv2D(1, kernel_size=5, padding='same', activation="sigmoid")(x)
+    rgb = Conv2D(3, kernel_size=5, padding='same', activation="tanh")(x)
+    out = concatenate([alpha, rgb])
+    return KerasModel(input_, out )    
+
+class Model():
+    IMAGE_SHAPE = (64, 64, 3)
+
+    def __init__(self, model_dir):
+
+        self.model_dir = model_dir
+
+        self.encoder = Encoder()
+        self.decoder_A = Decoder_ps()
+        self.decoder_B = Decoder_ps()
+
+        x = Input(shape=self.IMAGE_SHAPE)
+
+        self.netGA = KerasModel(x, self.decoder_A(self.encoder(x)))
+        self.netGB = KerasModel(x, self.decoder_B(self.encoder(x)))
+
+        self.netDA = Discriminator(nc_D_inp)
+        self.netDB = Discriminator(nc_D_inp)
+
+    def converter(self, swap):
+        converter = None
+        if not swap:
+            converter = Trainable(self.netGB, self.netDB, self.IMAGE_SHAPE)
+        else:
+            converter = Trainable(self.netGA, self.netDA, self.IMAGE_SHAPE)
+
+        return lambda img : converter.swap_face(img)
+
+    def load(self, swapped):
+        self.encoder.load_weights(ensure_file_exists(self.model_dir, encoderH5))

```

Comment:
Wrap all of the methods that load weights in a `try/except` so that it will attempt to load, but still continue and create the models if they do not exist.

Reply:
yep, I did this when integrating another PR, but I did this a bit brainlessly ;-) I'll revert...